src/sentry/utils/safe.py

```diff
@@ -122,16 +122,73 @@ def trim_dict(value, max_items=settings.SENTRY_MAX_DICTIONARY_ITEMS, **kwargs):
     return value
 
 
-def get_path(data, path, default=None):
+def get_path(data, *path, **kwargs):
     """
-    Looks up a path of properties in a nested dictionary safely.
-    Returns the value at the final level, or the default value if
-    property lookup failed at any step in the path.
+    Safely resolves data from a recursive data structure. A value is only
+    returned if the full path exists, otherwise ``None`` is returned.
+
+    If the ``default`` argument is specified, it is returned instead of ``None``.
+
+    If the ``filter`` argument is specified and the value is a list, it is
+    filtered with the given callback. Alternatively, pass ``True`` as filter to
+    only filter ``None`` values.
     """
-    if not isinstance(path, (list, tuple)) or len(path) == 0:
-        raise ValueError
     for p in path:
-        if not isinstance(data, collections.Mapping) or p not in data:
-            return default
+        if isinstance(data, collections.Mapping) and p in data:
+            data = data[p]
+        elif isinstance(data, (list, tuple)) and -len(data) <= p < len(data):
+            data = data[p]
+        else:
+            return kwargs.get('default')
+
+    f = kwargs.get('filter')
+    if f and data and isinstance(data, (list, tuple)):
+        data = list(filter((lambda x: x is not None) if f is True else f, data))
+
+    return data if data is not None else kwargs.get('default')
+
+
+def set_path(data, *path, **kwargs):
+    """
+    Recursively traverses or creates the specified path and sets the given value
+    argument. `None` is treated like a missing value. If a non-mapping is
```

Comment:
```suggestion
    argument. `None` is treated like a missing value. If a non-mapping item is
```

Reply:
Added in https://github.com/getsentry/sentry/pull/10733